libs/vkd3d/command.c

```diff
@@ -5796,16 +5796,25 @@ static void d3d12_command_list_clear_uav(struct d3d12_command_list *list, const
 
     if (d3d12_resource_is_buffer(resource))
     {
+        const struct vkd3d_bound_buffer_range *ranges = desc->heap->buffer_ranges.host_ptr;
+
         if (args->has_view)
         {
-            VkDeviceSize byte_count = args->u.view->format->byte_count
-                    ? args->u.view->format->byte_count
-                    : sizeof(uint32_t);  /* structured buffer */
-            full_rect.right = args->u.view->info.buffer.size / byte_count;
+            if ((list->device->bindless_state.flags & VKD3D_TYPED_OFFSET_BUFFER) != 0)
```

Comment:
Nit: we never really do the `!= 0` thing when testing bit masks, could be removed here as well for consistency.

Reply:
Fixed.